core/templates/dev/head/services/StateTopAnswersStatsService.js

```diff
@@ -0,0 +1,123 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for maintaining the statistics of the top answers for
+ * each state of an exploration.
+ */
+
+oppia.factory('StateTopAnswersStatsService', [
+  '$http', '$injector', 'AngularNameService', 'AnswerClassificationService',
+  'ExplorationContextService', 'ExplorationStatesService',
+  'UrlInterpolationService',
+  function(
+      $http, $injector, AngularNameService, AnswerClassificationService,
+      ExplorationContextService, ExplorationStatesService,
+      UrlInterpolationService) {
+    /**
+     * @typedef {Object} AnswerStats - A record for the statistics of a single
```

Comment:
What does this docstring pertain to?

Reply:
Ah, I reference the type in other docstrings of this source file. This also gives me auto-completions in my IDE, so it's helpful in more ways than one!